code/script.js

```diff
@@ -224,102 +228,135 @@ const generateBoard = () => {
+    } 
+    else {
+      alert(`No, the person doesn't have ${value} ${category}! Remove all people with ${value} ${category}`)
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value);
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+  // Function: redraw the board with the remaining people.
+  generateBoard();
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// Function: the user confirms its guess (stored in a variable ) and the function checkMyGuess is invoked. 
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  const playerGuess = confirm(`Are you sure you want to guess on ${personToConfirm}...? 🤔`) 
+  if (playerGuess) {
+   checkMyGuess(personToConfirm)   
+ }
 }
 
-// If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+ // Check if player's guess is correct, show message in the win-or-lose section accordingly
+const checkMyGuess = (personToCheck) => { 
+  if (personToCheck === secret.name) {
+    winOrLoseText.innerText = `...you got it! The secret person is indeed ${personToCheck} 👏`;
+  }
+  else {
+    winOrLoseText.innerText = `Incorrect guess unfortunately, the secret person is actually ${secret.name}!`;
+    winOrLoseSection.style.display = 'flex';
+  }
+  winOrLoseSection.style.display = 'flex'; //Show the win-or-lose section
+  board.style.display = 'none'  //Hide the game board
 }
 
-// Invokes the start function when website is loaded
-start()
 
-// All the event listeners
-restartButton.addEventListener('click', start)
+start() // Invokes the start function 
+
+
+// ALL THE EVENTLISTENERS
+// When the click event is triggered, the start function is invoked
+restartButton.addEventListener('click', start);
+
+//When the dropdown is changed, the selectQuestion-function is invoked. 
+questions.addEventListener('change', selectQuestion);
+
+filter.addEventListener('click', checkQuestion);
```

Comment:
This is fairly self-explanatory, but since you added such nice comments explaining the other event listeners, perhaps a comment could have been added here as well.